src/main/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepository.java

```diff
@@ -174,11 +189,20 @@ private RuntimeException mapException(Exception exception, ResourceNotFoundExcep
         if (exception instanceof HttpClientErrorException
             && ((HttpClientErrorException) exception).getRawStatusCode() == HttpStatus.NOT_FOUND.value()) {
             return resourceNotFoundException;
-        } else if (exception instanceof HttpClientErrorException
+        } else {
+            return mapException(exception, "getting");
+        }
+    }
+
+    private RuntimeException mapException(Exception exception, String processDescription) {
```

Comment:
This code reflects a similar process in DefaultCaseDefinitionRepository, although I have refactored this new controller to reduce the amount of code repetition.

To move it to the RestExceptionHandler would involve wrapping the error in a new exception (to include unique message information) then unravelling it in the RestExceptionHandler.  This may have made sense if it was to be used by many clients but for one would just feel like an overhead.  **So I have left as is.**